amp-beta-tester.php

```diff
@@ -0,0 +1,204 @@
+<?php
+/**
+ * Plugin Name: AMP Beta Tester
+ * Description: Opt-in to receive non-stable release builds for the AMP plugin.
+ * Plugin URI: https://amp-wp.org
+ * Author: AMP Project Contributors
+ * Author URI: https://github.com/ampproject/amp-wp/graphs/contributors
+ * Version: 0.1
+ * Text Domain: amp-beta-tester
+ * Domain Path: /languages/
+ * License: GPLv2 or later
+ *
+ * @package AMP Beta Tester
+ */
+
+namespace AMP_Beta_Tester;
+
+define( 'AMP__BETA_TESTER__DIR__', dirname( __FILE__ ) );
+define( 'AMP_PLUGIN_FILE', 'amp/amp.php' );
+
+// DEV_CODE. This block of code is removed during the build process.
+if ( file_exists( AMP__BETA_TESTER__DIR__ . '/amp.php' ) ) {
+	add_filter(
+		'site_transient_update_plugins',
+		function ( $updates ) {
+			if ( isset( $updates->response ) && is_array( $updates->response ) ) {
+				if ( array_key_exists( 'amp/amp-beta-tester.php', $updates->response ) ) {
+					unset( $updates->response['amp/amp-beta-tester.php'] );
+				}
+			}
+
+			return $updates;
+		}
+	);
+}
+register_activation_hook( __FILE__, __NAMESPACE__ . '\force_plugin_update_check' );
```

Comment:
The plugin activation hook could `activate_plugin( 'amp/amp.php' )` ([docs](https://developer.wordpress.org/reference/functions/activate_plugin/))

Reply:
Should it be activated programatically? There might be a scenario where the user wishes to keep the AMP plugin updated but not activated (for whatever reason). I think the notice telling the user that the AMP plugin must be activated is good enough.

Reply:
> There might be a scenario where the user wishes to keep the AMP plugin updated but not activated (for whatever reason).

Makes sense. But how is this doable when the beta tester plugin aborts early when the AMP plugin is not active?

Following that logic, my understanding is that in this case the beta tester plugin should work even when the AMP plugin is inactive, and should not show any warning.

Just my two cents while reading through the code :-)

Reply:
As far as I can tell the `register_activation_hook` hook would only be called upon plugin activation, while the `init` hook would be called upon every request, so it should show the notice regardless.

Reply:
Right. What I meant is that due to the early `return` in `init()`, the beta tester plugin does not work when the AMP plugin is not active.

However, above you mentioned that might be a valid use case:

> There might be a scenario where the user wishes to keep the AMP plugin updated but not activated

Correct me if I'm wrong :-)

Reply:
Oh, misunderstood your concern there. The status of the AMP plugin will not interfere with this one, as WP detects them as two separate plugins :smile:.

Reply:
Oh, misunderstood your concern. I'll modify `init` to accommodate that scenario :smile:.

Reply:
Here's the new logic:

https://github.com/ampproject/amp-wp/blob/ac056ebec93825bdf2725d30b0d2d0e852d8358d/amp-beta-tester.php#L55-L63